tests/server.py

```diff
@@ -251,6 +253,7 @@ def __init__(self):
         self._tcp_callbacks = {}
         self._udp = []
         lookups = self.lookups = {}
+        self._thread_callbacks = deque()  # type: Deque[Callable[[], None]]()
```

Comment:
Are you using a deque simply for the performance benefits of `popleft`?

Reply:
I'm using it because I don't think a `list` is threadsafe.

Reply:
Doing some research it looks to be threadsafe in CPython, however using a deque is likely a safer bet.